test/functional/feature_nulldummy.py

```diff
@@ -51,10 +51,17 @@ def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
     def run_test(self):
-        self.address = self.nodes[0].getnewaddress()
-        self.ms_address = self.nodes[0].addmultisigaddress(1, [self.address])['address']
-        self.wit_address = self.nodes[0].getnewaddress(address_type='p2sh-segwit')
-        self.wit_ms_address = self.nodes[0].addmultisigaddress(1, [self.address], '', 'p2sh-segwit')['address']
+        self.nodes[0].createwallet(wallet_name='wmulti', disable_private_keys=True)
+        wmulti = self.nodes[0].get_wallet_rpc('wmulti')
+        w0 = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.address = w0.getnewaddress()
+        self.pubkey = w0.getaddressinfo(self.address)['pubkey']
+        self.ms_address = wmulti.addmultisigaddress(1, [self.pubkey])['address']
+        self.wit_address = w0.getnewaddress(address_type='p2sh-segwit')
+        self.wit_ms_address = wmulti.addmultisigaddress(1, [self.pubkey], '', 'p2sh-segwit')['address']
+        if not self.options.descriptors:
```

Comment:
In commit "Use separate watchonly wallet for multisig in feature_nulldummy.py" (ed97c9b67e715815f653e20f5e478a4dad4e13f8)

Can a comment be added to explain why the non-descriptor version of the test needs these addresses imported to pass, while the descriptor version doesn't need them? (Assuming that is the case.) I'm missing some understanding here

Reply:
Legacy wallets need these imported for the test to work. Otherwise they won't watch the multisig. Descriptor wallets will always watch imported things.

Reply:
Added a comment.